perceval/utils/matrix.py

```diff
@@ -145,19 +146,44 @@ def random_unitary(n: int, parameters: Optional[Union[np.ndarray,list]] = None)
         :return: a numeric Matrix
         """
         if parameters is not None:
-            assert len(parameters) == 2*n**2, "parameters has not the right size: should be %d, and is %d" % (
-                2*n**2, len(parameters)
-            )
-            a = np.reshape(parameters[:n**2], (n, n))
-            b = np.reshape(parameters[n**2:], (n, n))
+            warnings.warn("use parametrized_unitary(parameters) instead to create a parametrized unitary "
+                          "matrix, version=0.11", DeprecationWarning)
+            assert len(parameters) == 2 * n ** 2, "parameters do not have the right size: should be %d, and is %d" % (
```

Comment:
If parameters is not None, raise the deprecated warning and call parametrized_unitary no ?

Reply:
Yes, I missed to think of that and re-wrote the code. Ooops